backend/crowdin/client.go

```diff
@@ -272,67 +265,223 @@ func (c *Client) Sync(ctx context.Context, d *directus.Handler) error {
+		err = c.syncAchievementGroups(ctx, TranslationHandler{
+			SaveTranslation: func(ctx context.Context, t SimpleTranslation) error {
+				return c.q.UpdateAchievementGroupTranslation(ctx, sqlc.UpdateAchievementGroupTranslationParams{
+					ItemID:      utils.AsUuid(t.ParentID),
+					Language:    t.Language,
+					Title:       null.StringFrom(t.Values["title"]),
+					Description: null.StringFrom(t.Values["description"]),
+				})
+			},
+		}, project, directory.ID, crowdinTranslations)
 		if err != nil {
 			return err
 		}
-		err = c.syncSurveys(ctx, d, project, directory.ID, crowdinTranslations)
+		err = c.syncSurveys(ctx, TranslationHandler{
+			SaveTranslation: func(ctx context.Context, t SimpleTranslation) error {
+				return c.q.UpdateSurveyTranslation(ctx, sqlc.UpdateSurveyTranslationParams{
+					ItemID:      utils.AsUuid(t.ParentID),
+					Language:    t.Language,
+					Title:       null.StringFrom(t.Values["title"]),
+					Description: null.StringFrom(t.Values["description"]),
+				})
+			},
+		}, project, directory.ID, crowdinTranslations)
 		if err != nil {
 			return err
 		}
-		err = c.syncSurveyQuestions(ctx, d, project, directory.ID, crowdinTranslations)
+		err = c.syncSurveyQuestions(ctx, TranslationHandler{
+			SaveTranslation: func(ctx context.Context, t SimpleTranslation) error {
+				return c.q.UpdateSurveyQuestionTranslation(ctx, sqlc.UpdateSurveyQuestionTranslationParams{
+					ItemID:      utils.AsUuid(t.ParentID),
+					Language:    t.Language,
+					Title:       null.StringFrom(t.Values["title"]),
+					Description: null.StringFrom(t.Values["description"]),
+				})
+			},
+		}, project, directory.ID, crowdinTranslations)
 		if err != nil {
 			return err
 		}
-		err = c.syncFAQs(ctx, d, project, directory.ID, crowdinTranslations)
+		err = c.syncFAQs(ctx, TranslationHandler{
+			SaveTranslation: func(ctx context.Context, t SimpleTranslation) error {
+				return c.q.UpdateFAQTranslation(ctx, sqlc.UpdateFAQTranslationParams{
+					ItemID:   utils.AsUuid(t.ParentID),
+					Language: t.Language,
+					Question: null.StringFrom(t.Values["question"]),
+					Answer:   null.StringFrom(t.Values["answer"]),
+				})
+			},
+		}, project, directory.ID, crowdinTranslations)
 		if err != nil {
 			return err
 		}
-		err = c.syncFAQCategories(ctx, d, project, directory.ID, crowdinTranslations)
+		err = c.syncFAQCategories(ctx, TranslationHandler{
+			SaveTranslation: func(ctx context.Context, t SimpleTranslation) error {
+				return c.q.UpdateFAQCategoryTranslation(ctx, sqlc.UpdateFAQCategoryTranslationParams{
+					ItemID:      utils.AsUuid(t.ParentID),
+					Language:    t.Language,
+					Title:       null.StringFrom(t.Values["title"]),
+					Description: null.StringFrom(t.Values["description"]),
+				})
+			},
+		}, project, directory.ID, crowdinTranslations)
 		if err != nil {
 			return err
 		}
-		err = c.syncGames(ctx, d, project, directory.ID, crowdinTranslations)
+		err = c.syncGames(ctx, TranslationHandler{
```

Comment:
Would it make sense for this to accept something like:

```
c.syncGames[T any](ctx, func(context, SimpleTranslation, func(ctx, T)), T, ...) 

// Call
c.syncGames(ctx, c.q.UpdateGameTranslation, sqlc.UpdateGameParams{}, ...)
```
Could just look more clean if it is legal in go